lib/g_searcher_web/controllers/helpers/search_helper.ex

```diff
@@ -0,0 +1,60 @@
+defmodule GSearcherWeb.Helpers.SearchHelper do
+  @valid_operators ["title", "url", "top_ads", "regular_ads", "status"]
+
+  @spec parse_query(String.t()) :: {:ok, %{String.t() => String.t()}}
+  def parse_query(query) when is_binary(query) do
+    parsed_query =
+      query
+      |> escape_percentage_sign()
+      |> split_string_by_space(ignore_in_double_quotes: true)
+      |> convert_search_options_to_map()
+      |> build_query_attributes()
+
+    {:ok, parsed_query}
+  end
+
+  def parse_query(nil), do: {:error, :no_query}
+
+  defp split_string_by_space(query, ignore_in_double_quotes: true) when is_binary(query),
+    do: Regex.split(~r{\s+(?=([^"]*"[^"]*")*[^"]*$)}, query)
```

Comment:
How about moving this to the module attribute? with the name, it easier to understand.